playground/src/components-provider/components-provider.js

```diff
@@ -21,25 +21,23 @@ export function createComponentsProvider() {
 class ComponentsProvider {
 
   constructor(win) {
-    this.componentsMap = {};
     this.win = win;
-    this.win.requestIdleCallback(() => this._fetchComponents());
+    this.componentsMap = new Promise((resolve, reject) => {
+      this.win.requestIdleCallback(() => {
+        let request = new Request(COMPONENTS_URL, {
+          headers: new Headers({'x-requested-by': 'playground'})
+        });
+        fetch(request)
+            .then((r) => r.json())
+            .then((data) => {
+              resolve(data);
+            });
```

Comment:
What happens if the fetch fails?

Reply:
How about a number of backoff retries, then ultimately if there is no success after n tries, assign the the components map to just be empty? What do you think?

Reply:
As it's not a critical feature I think it's fine to simply fail and fallback to an empty map at first. We can add a retry later.

Reply:
Sure. Done